PerfDiagnosticListener/PerfDiagnosticListener.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace PerfDiagnosticListener
+{
+    public class PerfDiagnosticListener
+    {
+        private static long _requestsBegan = 0;
+        private static long _requestsEnded = 0;
+        private static long _requestTicks = 0;
+
+        [DiagnosticName("Microsoft.AspNetCore.Hosting.BeginRequest")]
+        public void OnBeginRequest(HttpContext httpContext, long timestamp)
+        {
+            Interlocked.Increment(ref _requestsBegan);
+
+            var requestInfo = new RequestInfo { BeginTimestamp = timestamp };
+            httpContext.Features.Set(requestInfo);
```

Comment:
Reading the docs, I get the impression that `HttpContext.Features` is more intended for public stuff which may be used by multiple middlewares or app code, while `HttpContext.Items` is more intended for private stuff used by a single middleware or app code.  If true, this means `HttpContext.Items` is more appropriate in this case.  But I agree this is not clear.